### PR TITLE
Create service2initd

### PR DESCRIPTION
**Pull request by rizalmart**
This script generates init.d script from systemd .service file. Because some distros are no longer provides init.d scripts.

### CODE REVIEW

**Comment by 01micko** [comment]
I've asked @dimkr to review this PR as I don't use any systemd distro and am not familiar with the file formats.

**Comment by 01micko** [comment]
@dimkr ?

File: woof-code/rootfs-skeleton/usr/bin/service2initd [hunk 1]
```diff
@@ -0,0 +1,125 @@
+#!/bin/bash
+#Converts systemd .service file into init.d script
+#Written by mistfire
+
+SRVFILE="$1"
```
Comment (dimkr) [inline woof-code/rootfs-skeleton/usr/bin/service2initd:5]: There's a messy mix of spaces and tabs in this file

File: woof-code/rootfs-skeleton/usr/bin/service2initd [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+if [ "$SRVFILE" == "" ] || [ ! -e "$SRVFILE" ]; then
+ exit
```
Comment (dimkr) [inline woof-code/rootfs-skeleton/usr/bin/service2initd:8]: exit 1?

File: woof-code/rootfs-skeleton/usr/bin/service2initd [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+	 EXECCMD="$(cat $SRVFILE | grep '^ExecStart=' | sed -e 's#^ExecStart=##' | xargs)"
+	 PREEXEC="$(cat $SRVFILE | grep '^ExecStartPre=' | sed -e 's#^ExecStartPre=##' | xargs)"
+	 POSTEXEC="$(cat $SRVFILE | grep '^ExecStartPost=' | sed -e 's#^ExecStartPost=##' | xargs)"
+	 EXECSTOP="$(cat $SRVFILE | grep '^ExecStop=' | sed -e 's#^ExecStop=##' | xargs)"
+	 POSTEXECSTOP="$(cat $SRVFILE | grep '^ExecStopPost=' | sed -e 's#^ExecStopPost=##' | xargs)"
+	 RELOADEXEC="$(cat $SRVFILE | grep '^ExecReload=' | sed -e 's#^ExecReload=##' | xargs)"
+
+	 PIDFILE="$(cat $SRVFILE | grep '^PIDFile=' | cut -f 2 -d '=' | xargs)"
+	 KILLSIG="$(cat $SRVFILE | grep '^KillSignal=' | cut -f 2 -d '=' | xargs)"
+	 WKDIR="$(cat $SRVFILE | grep '^WorkingDirectory=' | cut -f 2 -d '=' | xargs)"
```
Comment (dimkr) [inline woof-code/rootfs-skeleton/usr/bin/service2initd:26]: This list of fields is partial - for example, Restart= is not taken care of and doesn't translate well to sysvinit

File: woof-code/rootfs-skeleton/usr/bin/service2initd [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+    #if [ "$ENVFILE" != "" ] && [ -e "$ENVFILE" ]; then
+	 echo ' . '$ENVFILE
+    #fi
+	
+    [ "$WKDIR" != "" ] && echo ' cd '$WKDIR
+    [ "$PREEXEC" != "" ] && echo ' '$PREEXEC
+    
+    if [ "$EXECCMD" != "" ]; then
+     if [ "$ENV_VALUES" != "" ]; then
+      echo ' '$ENV_VALUES' '$EXECCMD''$BCKPID
```
Comment (dimkr) [inline woof-code/rootfs-skeleton/usr/bin/service2initd:61]: Are you assuming that EXECCMD will daemonize? In the systemd world, not daemonizing (so systemd can restart you) is the norm

File: woof-code/rootfs-skeleton/usr/bin/service2initd [hunk 5]
```diff
@@ -0,0 +1,125 @@
+#!/bin/bash
```
Comment (dimkr) [inline woof-code/rootfs-skeleton/usr/bin/service2initd:1]: What uses this script? What is the % of successful translation?

**Review by dimkr — CHANGES_REQUESTED** [review]

**Pull request closed without merging**